test/memspaces/memspace_host_all.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2024 Intel Corporation
+// Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "memory_target_numa.h"
+#include "memspace_helpers.hpp"
+#include "memspace_internal.h"
+#include "test_helpers.h"
+
+#include <numa.h>
+#include <numaif.h>
+#include <umf/memspace.h>
+
+using umf_test::test;
+
+TEST_F(numaNodesTest, memspaceGet) {
+    umf_memspace_handle_t hMemspace = umfMemspaceHostAllGet();
+    UT_ASSERTne(hMemspace, nullptr);
+
+    // Confirm that the HOST ALL memspace is composed of all available NUMA nodes.
+    UT_ASSERTeq(hMemspace->size, nodeIds.size());
+    for (size_t i = 0; i < hMemspace->size; i++) {
+        // NUMA memory target internally casts the config directly into priv.
+        struct umf_numa_memory_target_config_t *numaTargetCfg =
```

Comment:
please add a TODO here: "use the memory target API when it would be available"

Reply:
Done.